pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,153 @@
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        self.write_config_console(None)
+
+        # Rewrite the loader.conf
+        # For now we are just updating the timeout to actually
+        # implement the bootloader --timeout option
+        config_path = "%s%s" % (conf.target.system_root, self.config_file)
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        config = open(config_path, "w")
+        config.write("timeout "+ str(self.timeout) + "\n")
+        config.write("#console-mode keep\n")
+        config.close()
+
+        # update /etc/kernel/cmdline
+        # should look something like "root=UUID=45b931b7-592a-46dc-9c33-d38d5901ec29 ro resume=/dev/sda3"
+        config_path = "%s%s" % (conf.target.system_root, "/etc/kernel/cmdline")
+        log.info("systemd.py: write_config systemd commandline : %s ", config_path)
+        config = open(config_path, "w")
+        args = " ".join(list(self.boot_args._arguments))
+        log.info("systemd.py: systemd used boot args: %s ", args)
+
+        # pick up the UUID of the mounted rootfs,
+        root_uuid = util.execWithCapture("findmnt", [ "-sfn", "-oUUID", "/" ],
+                                          root=conf.target.system_root)
+        args += " root=UUID=" + root_uuid
+        config.write(args)
+        config.close()
+        # rather than creating a mess in python lets just
+        # write the options above, and run a script which will merge the
+        # boot cmdline (after stripping inst. and BOOT_) with the anaconda
+        # settings and the kernel-install recovery/etc options.
+        rc = util.execWithRedirect(
+            "/usr/sbin/updateloaderentries",
+            [" "],
+            root=conf.target.system_root
+        )
+        if rc:
+            raise BootLoaderError("failed to write boot loader configuration")
+
+    #
+    # installation
+    #
+    def install(self, args=None):
+        log.info("systemd.py: install")
+        if args is None:
+            args = []
+
+        log.info("systemd.py: install systemd boot install (root=%s)", conf.target.system_root)
+
+        # the --esp-path= isn't strictly required, but we want to be explicit about it.
+        rc = util.execWithRedirect("bootctl", [ "install", "--esp-path=/boot/efi",
+                                                "--efi-boot-option-description=" + productName.split("-")[0] ],
+                                   root=conf.target.system_root,
+                                   env_prune=['MALLOC_PERTURB_'])
+        if rc:
+            raise BootLoaderError("boot loader install failed")
```

Comment:
Same here.

Reply:
I'm a bit at a loss for what might cause a failure like this (and would dedicate error messages if I could accurately guess what might cause them, possible ESP capacity/format, or uefi variable storage issues? Or just things missing from the initramfs?)  

This is what I've come up with:

"bootctl failed to install UEFI boot loader. "
"More information may be found in /tmp/program.log,"
"/tmp/storage.log or /tmp/anaconda.log")